### PR TITLE
Add handling for Temporal types

### PR DESCRIPTION
**Pull request by cn-stephen**
Temporal types throw on valueOf/Symbol.toPrimitive, so the existing coercion-based comparison breaks. Detect them by constructor name and compare via String() instead.

- Rename objKey/newObjKey to value/newValue for clarity
- Add temporalTypes lookup for all 8 Temporal types
- Gate Object.getPrototypeOf behind areCompatibleObjects for perf
- Guard against null values in objConstructor computation
- Use early-exit continue pattern for clearer control flow
- Add richTypes guard to coercion comparison for explicitness
- Add tests for all Temporal types including cross-type replacement
- Add regression test for null-to-object replacement

### CODE REVIEW

File: tests/basic.js [hunk 1]
```diff
@@ -42,6 +42,17 @@ test("replace object with null", () => {
 	]);
 });
 
+test("replace null with object", () => {
```
Comment (AsyncBanana) [inline tests/basic.js:45]: In the future, please isolate these sorts of changes that aren't absolutely required into their own PRs

File: index.ts [hunk 2]
```diff
@@ -35,56 +46,79 @@ export default function diff(
 	const isObjArray = Array.isArray(obj);
 
 	for (const key in obj) {
-		const objKey = obj[key];
+		const value = obj[key];
```
Comment (AsyncBanana) [inline index.ts:49]: See the earlier comment. I agree this is a good change, but the combination of multiple changes makes it more difficult to review.

**Review by AsyncBanana — APPROVED** [review]
There are a few changes to make things a little more concise that I will add before release, and, in the future, please separate your pull requests a little more, but it looks good!

File: tests/basic.js (hunk 1 above)
Comment (cn-stephen) [reply tests/basic.js:45]: will do

File: index.ts (hunk 2 above)
Comment (cn-stephen) [reply index.ts:49]: sorry, you're right
